### PR TITLE
Hydration wip

### PR DESCRIPTION
**Pull request by coleanthony1990**
Created a class for hydration. all methods will need to be invoked in scripts with args as the inputs (i.e. user or date or data). still brainstorming how to get the dates to connect to the indexes for showing a weeks worth of data.

## Type of change

- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)
- [ ] Refactor

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [ ] I tested my changes in the browser
- [x] I tested with Mocha/Chai

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new bugs
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Is your code D.R.Y?
- [x] Does it follow SRP?

### CODE REVIEW

File: src/Hydration.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    getAvgOunces(userInfo) {
+      const totalOunces = userInfo.reduce((acc, entry) => {
+        if (this.id === entry.userID) {
+          acc += entry.numOunces;
+        }
+        return acc;      
+      }, 0);   
+      return Math.floor(totalOunces / this.ounces.length);
+    }
+      }
```
Comment (ohClaire) [inline src/Hydration.js]: align indentation

File: src/Hydration.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    //     new User(user)
+      
+    //   console.log(userAvgOunces)
+    //   return userAvgOunces
+    // }, [])
+/*We want create an array where the keys are user ids and the avg Ounces. the value of avg
+ounces is avg ounces per user. 
+iterate through hydrationData to get id and then another iteration to get get numOunces.
+*/
+
```
Comment (ohClaire) [inline src/Hydration.js]: remove anotations? You can save your pseudocode in our group project Notion

File: test/Hydration-test.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    expect(hydrate1.ouncesPerDay('2019/06/16', hydrationData)).to.deep.equal({'2019/06/16': 69})
+  })
+  it ('should calculate weekly ouces per user', () => {
+    expect(hydrate1.getDailyOuncesByWeek(hydrationData, 0, 2)).to.deep.equal([
+      { userID: 1, date: '2019/06/15', numOunces: 37 },
+      { userID: 1, date: '2019/06/16', numOunces: 69 }
+    ])
+  })
+  it('should calculate the avg ounces per user', () => {
+    expect(hydrate1.getAvgOunces(hydrationData)).to.deep.equal(59)
```
Comment (ohClaire) [inline test/Hydration-test.js]: semicolons

File: test/Hydration-test.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import Hydration from '../src/Hydration';
+import hydrationData from "../src/data/sample-hydration";
+import userData from '../src/data/users.js'
+
+describe('Hydration', () => {
+  let hydrate1;
+  let hydrate2;
+  
+  beforeEach(() => {
+    hydrate1 = new Hydration(1, hydrationData)
```
Comment (ohClaire) [inline test/Hydration-test.js]: semicolons

**Review by ohClaire — CHANGES_REQUESTED** [review]
Looks great! There were some semicolons missing. Give the Hydration-test a look over as well to add semicolons.

File: src/Hydration.js (hunk 2 above)
Comment (coleanthony1990) [reply src/Hydration.js]: I just deleted it. It was mostly just code that i was trying and then commented out.

**Review by ohClaire — APPROVED** [review]

**Pull request merged**